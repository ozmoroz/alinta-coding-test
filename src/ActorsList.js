// @flow
import * as React from 'react';
import type { ActorData } from './types';

type Props = {
  actors: ?Array<ActorData>
};

/**
 * Render an info for a single actor
 */
const renderActor = (actor: ActorData) => (
  <li key={actor.actor}>
    {actor.actor}
    <ul>
      {actor.roles.map((role, roleIndex) => <li key={roleIndex}>{role}</li>)}
    </ul>
  </li>
);

const ActorsList = (props: Props) => {
  if (!props.actors) return null;
  return (
    <ul className="list-unstyled">
      {props.actors.map(actor => renderActor(actor))}
    </ul>
  );
};

export default ActorsList;
