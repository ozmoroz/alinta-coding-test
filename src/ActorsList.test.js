import React from 'react';
import renderer from 'react-test-renderer';
import ActorsList from './ActorsList';

describe('ActorsList', () => {
  const actors = [
    { actor: 'Alex Borstein', roles: ['Lois Griffin'] },
    { actor: 'Patrick Warburton', roles: ['Joe Swanson'] },
    { actor: 'Seth Green', roles: ['Luke Skywalker', 'Chris Griffin'] },
    { actor: 'Mila Kunis', roles: ['Meg Griffin'] },
    { actor: 'Keifer Sutherland', roles: ['Dr Barry Wolfson', 'Ace Merrill'] },
    { actor: 'Richard Dreyfuss', roles: ['The Writer'] },
    { actor: 'River Phoenix', roles: ['Chris Chambers'] },
    { actor: 'Wil Wheaton', roles: ['Gorgie Lachance', 'Romulan'] },
    { actor: 'Corey Feldman', roles: ['Teddy Duchamp'] },
    { actor: 'Winona Ryder', roles: ['Amanda Grayson'] },
    { actor: 'Leonard Nimoy', roles: ['Spock'] },
    { actor: 'Eric Bana', roles: ['Nero'] },
    { actor: 'Chris Pine', roles: ['Kirk'] },
    { actor: 'Simon Pegg', roles: ['Scotty'] }
  ];

  it('renders correctly', () => {
    const tree = renderer.create(<ActorsList actors={actors} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
