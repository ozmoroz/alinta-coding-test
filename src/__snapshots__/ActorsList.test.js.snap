// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorsList renders correctly 1`] = `
<ul
  className="list-unstyled"
>
  <li>
    Alex Borstein
    <ul>
      <li>
        Lois Griffin
      </li>
    </ul>
  </li>
  <li>
    Patrick Warburton
    <ul>
      <li>
        Joe Swanson
      </li>
    </ul>
  </li>
  <li>
    Seth Green
    <ul>
      <li>
        Luke Skywalker
      </li>
      <li>
        Chris Griffin
      </li>
    </ul>
  </li>
  <li>
    Mila Kunis
    <ul>
      <li>
        Meg Griffin
      </li>
    </ul>
  </li>
  <li>
    Keifer Sutherland
    <ul>
      <li>
        Dr Barry Wolfson
      </li>
      <li>
        Ace Merrill
      </li>
    </ul>
  </li>
  <li>
    Richard Dreyfuss
    <ul>
      <li>
        The Writer
      </li>
    </ul>
  </li>
  <li>
    River Phoenix
    <ul>
      <li>
        Chris Chambers
      </li>
    </ul>
  </li>
  <li>
    Wil Wheaton
    <ul>
      <li>
        Gorgie Lachance
      </li>
      <li>
        Romulan
      </li>
    </ul>
  </li>
  <li>
    Corey Feldman
    <ul>
      <li>
        Teddy Duchamp
      </li>
    </ul>
  </li>
  <li>
    Winona Ryder
    <ul>
      <li>
        Amanda Grayson
      </li>
    </ul>
  </li>
  <li>
    Leonard Nimoy
    <ul>
      <li>
        Spock
      </li>
    </ul>
  </li>
  <li>
    Eric Bana
    <ul>
      <li>
        Nero
      </li>
    </ul>
  </li>
  <li>
    Chris Pine
    <ul>
      <li>
        Kirk
      </li>
    </ul>
  </li>
  <li>
    Simon Pegg
    <ul>
      <li>
        Scotty
      </li>
    </ul>
  </li>
</ul>
`;
